orttraining/orttraining/eager/opgen/opgen/generator.py

```diff
@@ -254,6 +254,16 @@ def _write_function_body(
     onnx_op.eval(ctx)
     ctx.prepare_outputs()
 
+    # Fetch the ORT invoker from an at::Tensor.device()
+    # FIXME: find the first at::Tensor param anywhere in the signature
+    # instead of simply the first parameter?
+    first_param = cpp_func.parameters[0].member
```

Comment:
will the TensorList always be the first parameter?

Reply:
The whole code is written on the assumption of the first parameter being the tensor, the comment and the line were there already I just moved it a few lines above. The changes I have made will handle an additional case of when the first parameter is a TensorList. I can look into removing the assumption of first parameter not being tensor or tensorlist, let me know?

Reply:
hmm... that code looks buggy. anyway, put a TODO there let's make it more robust later.